caikit/core/model_management/local_model_trainer.py

```diff
@@ -0,0 +1,328 @@
+            """Every model future must be able to poll the status of the
+            training job
+            """
+            # The worker was canceled while doing work. It may still be in the
+            # process of terminating and thus still alive.
+            if self._worker.canceled:
+                return ModelTrainerBase.TrainingStatus.CANCELED
+
+            # If the worker is currently alive it's doing work
+            if self._worker.is_alive():
+                return ModelTrainerBase.TrainingStatus.RUNNING
+
+            # The worker threw outside of a cancellation process
+            if self._worker.threw:
+                return ModelTrainerBase.TrainingStatus.ERRORED
+
+            # The worker completed its work without being canceled or raising
+            if self._worker.ran:
+                return ModelTrainerBase.TrainingStatus.COMPLETED
+
+            # If it's not alive and not done, it hasn't started yet
+            return ModelTrainerBase.TrainingStatus.QUEUED
+
+        def cancel(self):
+            """Terminate the given training"""
+            log.debug("Canceling training %s", self.id)
+            with alog.ContextTimer(
+                log.debug2, "Done canceling training %s in: ", self.id
+            ):
+                log.debug3("Destroying worker in %s", self.id)
+                self._worker.destroy()
+
+        def wait(self):
+            """Block until the job reaches a terminal state"""
+            log.debug2("Waiting for %s", self.id)
+            self._worker.join()
+            log.debug2("Done waiting for %s", self.id)
+
+            # If running a subprocess, handle abnormal exit codes
+            if self._use_subprocess:
+                self._completion_time = self._worker.get_or_throw()
+                if self._worker.exitcode and self._worker.exitcode != os.EX_OK:
+                    if self._worker.exitcode == OOM_EXIT_CODE:
+                        raise MemoryError("Training process died with OOM error!")
+                    if not self._worker.canceled:
+                        raise RuntimeError(
+                            f"Training process died with exit code {self._worker.exitcode}"
+                        )
+
+        def load(self) -> ModuleBase:
+            """Wait for the training to complete, then return the resulting
+            model or raise any errors that happened during training.
+            """
+            self.wait()
+            if self._use_subprocess:
+                log.debug2("Loading model saved in subprocess")
+                error.value_check(
+                    "<COR16745216E>",
+                    self.save_path,
+                    "Unable to load model trained in subprocess without a save_path",
+                )
+                error.value_check(
+                    "<COR59551640E>",
+                    os.path.exists(self.save_path),
+                    "Unable to load model saved in subprocess at {}",
```

Comment:
If the path doesn't exist I might expect a slightly more specific error message, maybe
"Unable to load model saved in subprocess, path does not exist: {}"

Reply:
Also, should all of these logs contain `self.id` for traceability?